File: src/main/java/com/leijendary/spring/boot/template/util/StringUtil.java
package com.leijendary.spring.boot.template.util;

import static java.lang.Character.toLowerCase;
import static java.lang.Character.toUpperCase;

public class StringUtil {

    public static String snakeCaseToCamelCase(final String value) {
        return snakeCaseToCamelCase(value, false);
    }

    public static String snakeCaseToCamelCase(final String value, final boolean capitalizeFirst) {
        final var builder = new StringBuilder();
        final var strings = value.split("_");

        for (final String string: strings) {
            builder.append(upperCaseFirst(string));
        }

        final var result = builder.toString();

        if (!capitalizeFirst) {
            return lowerCaseFirst(result);
        }

        return result;
    }

    public static String upperCaseFirst(final String value) {
        final char[] chars = value.toCharArray();
        chars[0] = toUpperCase(chars[0]);

        return new String(chars);
    }

    public static String lowerCaseFirst(final String value) {
        final char[] chars = value.toCharArray();
        chars[0] = toLowerCase(chars[0]);

        return new String(chars);
    }
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/util/StringUtil.kt
package com.leijendary.spring.boot.template.core.util

import java.lang.Character.toLowerCase
import java.lang.Character.toUpperCase

fun String.snakeCaseToCamelCase(capitalizeFirst: Boolean = false): String {
    val builder = StringBuilder()
    val strings = this.split("_".toRegex()).toTypedArray()
    strings.forEach { builder.append(it.upperCaseFirst()) }

    val result = builder.toString()

    return if (!capitalizeFirst) result.lowerCaseFirst() else result
}

fun String.upperCaseFirst(): String {
    val chars = this.toCharArray()
    chars[0] = toUpperCase(chars[0])

    return String(chars)
}

fun String.lowerCaseFirst(): String {
    val chars = this.toCharArray()
    chars[0] = toLowerCase(chars[0])

    return String(chars)
}

fun String.isInt(): Boolean {
    return try {
        this.toInt()
        true
    } catch (ignored: NumberFormatException) {
        false
    }
}

fun String.isLong(): Boolean {
    return try {
        this.toLong()
        true
    } catch (ignored: NumberFormatException) {
        false
    }
}